notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+
+
+\section{Lecture 22: Non-convex constraints} 
+
+\begin{center}
+Seyed Mehran Mirramezani and Serena Yuan
+\end{center}
+
+
+Last time we discussed optimization of a convex function $f$ over non-convex sets. A typical example for such a problem is $\min \| Ax-y\|_2^2$ subject to $\|x\|_0 \leq s$. Another type of non-convex optimization is to minimize a non-convex function over a convex set. A typical example for this kind of non-convex optimization is known as $l_0$-minimization where $\min \| x\|_0$ subject to an affine convex constraint like $Ax=y$. In sum, the $l_0$-minimization is: 
+
+$$\min_{Ax=y} \|x\|_0$$
+
+One option for solving the above non-convex optimization problem is to relax the $l_0$-objective to the convex $l_1$-objective as:
+
+$$\min_{Ax=y} \|x\|_1$$
+
+A variation of the above problem is Basis Pursuit Denoising (BPDN),
+
+\begin{align}
+\min \|x\|_1 \text{ s.t. } \|Ax - y \|_2 \leq \epsilon.
+\end{align}
+
+
+Note that under appropriate assumptions on $A$, $x$ and $y$, we observed that $l_1$-minimization still gives the right answer (RIP, restricted nullspace).
+Sparse linear regression has received a lot of attention and there are many related optimization programs like:
+BPDN (allows for noise)
+$$\min_{\|Ax-y\|_2 \leq s} \|x\|_1$$
+
+Constrained Lasso
+$$\min_{\|x\|_1 \leq \lambda }\|Ax-y\|_2^2 $$
+
+Lagrangian Lasso
+$$\min \|Ax-y\|_2^2 +  \lambda \|x\|_1$$
+
+In our setting, we have \\
+
+\centering{nonconvex constraint} 
+$$\downarrow$$
+\centering{cvx relaxation }
+$$\downarrow$$
+\centering{PGD (Projected Gradient Descent)},
+
+or more directly, nonconvex constraint $\rightarrow$ PGD. 
+
+Here, the Restricted Isometry Property allows optimization over nonconvex sets.
+
+
+Now, we discuss the Iterative Hard Thresholding (IHT) which is known as a projected gradient descent (PGD) for sparse vectors.
+
+Iterative Hard Thresholding (IHT) is used with Gradient Descent for the following problem,
+
+\begin{align}
+f(x) &= \frac{1}{2} \| Ax - y \|_2^2 \\
+\nabla f(x) &= A^{\trans}(Ax - y)
+\end{align}
+
+If it is possible to directly optimize over nonconvex sets, then we don't need convex relaxation.
+
+We want to show that the algorithm IHT outputs a solution. The algorithm is defined as:
+
+\begin{align*}
+\text{IHT}(y, A, t, s) \\
+x^1 &\leftarrow 0 \\
+\text{ for } &i=1 ,\cdots, t, \\
+\tilde{x}^{i+1} &\leftarrow x^i - A^{\trans} (Ax - y) \\
+x^{i+1} &\leftarrow P_S( \tilde{x}^{i+1} ) \\
+\text{ return }& \hat{x} \leftarrow x^{t+1}
+\end{align*}
+
+Note that $P_s$ is a projection onto a set of s-sparse vectors. In the rest of the lecture we will see how we should do this projection and how fast the method is.
+
+RIP: matrix $A$ satisfies the $(s,\delta)$-RIP if for all $s$-sparse vectors we have:
+$$(1-\delta) \|x\|_2^2 \leq \|Ax\|_2^2 \leq (1+\delta)\|x\|_2^2$$
```

Comment:
It would be nice to have a sentence here saying why we introduce RIP.

Also, regarding "$$" in LaTeX: Generally \[ and \] is preferable, e.g., see

https://tex.stackexchange.com/questions/503/why-is-preferable-to